lti_consumer/tests/unit/test_lti.py

```diff
@@ -190,7 +217,101 @@ def test_get_signed_lti_parameters(self):
+        del expected_lti_parameters['custom_student_verification_status']
+        self.assertEqual(self.lti_consumer.get_signed_lti_parameters(), expected_lti_parameters)
+
+    @patch(
+        'lti_consumer.lti.get_oauth_request_signature',
+        Mock(return_value=(
+            'OAuth oauth_nonce="fake_nonce", '
+            'oauth_timestamp="fake_timestamp", oauth_version="fake_version", oauth_signature_method="fake_method", '
+            'oauth_consumer_key="fake_consumer_key", oauth_signature="fake_signature"'
+        ))
+    )
+    @patch(
+        'lti_consumer.lti_consumer.LtiConsumerXBlock.prefixed_custom_parameters',
+        PropertyMock(return_value={u'custom_param_1': 'custom1', u'custom_param_2': 'custom2'})
+    )
+    @patch(
+        'lti_consumer.lti_consumer.LtiConsumerXBlock.lti_provider_key_secret',
+        PropertyMock(return_value=('t', 's'))
+    )
+    @patch('lti_consumer.lti_consumer.LtiConsumerXBlock.user_id', PropertyMock(return_value=FAKE_USER_ID))
+    def test_get_signed_lti_parameters_no_course_mode(self):
+        """
+        Test `get_signed_lti_parameters` returns the correct dict when the author does not wish to
+        transmit course_mode and verification status
+        """
+        self.lti_consumer.xblock.due = timezone.now()
+        self.lti_consumer.xblock.graceperiod = timedelta(days=1)
+
+        expected_lti_parameters = self._get_base_expected_lti_parameters()
+        self.lti_consumer.xblock.has_score = True
+        self.lti_consumer.xblock.ask_to_send_username = True
+        self.lti_consumer.xblock.ask_to_send_email = True
+        self.lti_consumer.xblock.transmit_course_mode_and_status = False
+        self.lti_consumer.xblock.runtime.get_real_user.return_value = Mock(
+            email='edx@example.com',
+            username='edx',
+            preferences=Mock(filter=Mock(return_value=[Mock(value='en')]))
+        )
+
+        self.lti_consumer.xblock.runtime.service = _mock_services
+
+        self.assertEqual(self.lti_consumer.get_signed_lti_parameters(), expected_lti_parameters)
+
+    @patch(
+        'lti_consumer.lti.get_oauth_request_signature',
+        Mock(return_value=(
+            'OAuth oauth_nonce="fake_nonce", '
+            'oauth_timestamp="fake_timestamp", oauth_version="fake_version", oauth_signature_method="fake_method", '
+            'oauth_consumer_key="fake_consumer_key", oauth_signature="fake_signature"'
+        ))
+    )
+    @patch(
+        'lti_consumer.lti_consumer.LtiConsumerXBlock.prefixed_custom_parameters',
+        PropertyMock(return_value={u'custom_param_1': 'custom1', u'custom_param_2': 'custom2'})
+    )
+    @patch(
+        'lti_consumer.lti_consumer.LtiConsumerXBlock.lti_provider_key_secret',
+        PropertyMock(return_value=('t', 's'))
+    )
+    @patch('lti_consumer.lti_consumer.LtiConsumerXBlock.user_id', PropertyMock(return_value=FAKE_USER_ID))
+    def test_get_signed_lti_parameters_none_verification_status(self):
```

Comment:
@chrisndodge: You have a couple of options:

1. `@patch` the class (which will apply the same decorators to every test method)
2. Write your own decorator that bundles all of the existing decorators:

    def decorators_for_tests(func):
        func = decorator1(args)(func)
        func = decorator2(args)(func)
        ...
        return func